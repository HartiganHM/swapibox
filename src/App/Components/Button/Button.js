import React from 'react';
import PropTypes from 'prop-types';
import './Button.css';

const Button = ({ type, value, selectData, currentDisplay, favoriteCount }) => {
  const buttonClass =
    value.toLowerCase() === currentDisplay ? `${type} selected` : type;
  const buttonImageSrc = '../../../Images/Icons/' + value.toLowerCase() + '.svg'
  const buttonImage = <img className='category-button-image' src={buttonImageSrc} />;

  return (
    <button className={buttonClass} onClick={() => selectData(value)}>
      {value}{' '}
      <span className='counter'>
        {value === 'View Favorites' ? `${favoriteCount}` : buttonImage}
      </span>
    </button>
  );
};

Button.propTypes = {
  type: PropTypes.string,
  value: PropTypes.string,
  selectData: PropTypes.func,
  currentDisplay: PropTypes.string,
  favoriteCount: PropTypes.number
};

export default Button;
