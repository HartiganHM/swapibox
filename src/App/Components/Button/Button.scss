@import '../../../variables.scss';

.button {
    border: none;
    padding: $padding-small;
}

.category {
    align-items: center;
    background-color: $button-category;
    border: 2px solid $white;
    border-radius: 0.3rem;
    color: $white;
    display: flex;
    font-size: 1.8rem;
    justify-content: space-around;
    transition: background-color 0.1s ease;
    width: 15rem;

    &:hover {
        background-color: $button-hover;
    }

    .category-button-image-people,
    .category-button-image-planets,
    .category-button-image-vehicles {
        background-position: center;
        background-repeat: no-repeat;
        background-size: 3rem;
        width: 3rem;
        height: 3rem;
    }

    .category-button-image-people {
        background-image: url('../../../Images/Icons/people-selected.svg');
    }

    .category-button-image-planets {
        background-image: url('../../../Images/Icons/planets-selected.svg');
    }

    .category-button-image-vehicles {
        background-image: url('../../../Images/Icons/vehicles-selected.svg');
    }
}

.selected {
    background-color: $background-dark;
}

.favorites {
    background-color: $white;
    position: absolute;
    top: 4rem;
    right: 4rem;

    .counter {
        background-color: $background-dark;
        color: $white;
        padding: 0.5rem;
    }
}