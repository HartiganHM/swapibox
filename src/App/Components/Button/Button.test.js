import React from 'react';
import Button from './Button';
import { shallow } from 'enzyme';

describe('Button Tests', () => {
  let renderedButton;
  let mockType;
  let mockValue;
  let mockSelectData;
  let mockCurrentDisplay;

  beforeEach(() => {
    mockType = 'category button';
    mockValue = 'People';
    mockSelectData = jest.fn();
    mockCurrentDisplay = '';

    renderedButton = shallow(
      <Button
        type={mockType}
        value={mockValue}
        selectData={mockSelectData}
        currentDisplay={mockCurrentDisplay}
      />
    );
  });

  it('Should exist', () => {
    expect(renderedButton).toBeDefined();
  });

  it('Should match the snapshot', () => {
    expect(renderedButton).toMatchSnapshot();
  });

  it('Should have a matching image based on the incoming value', () => {
    expect(renderedButton.find('.category-button-image-people').length)
    .toEqual(1);

    mockValue = 'Vehicles';
    renderedButton = shallow(<Button value={mockValue} />);

    expect(
      renderedButton.find('.category-button-image-vehicles').length)
      .toEqual(1);
  });

  it('Should have a counter if the incoming value is View Favorites', () => {
    mockValue = 'View Favorites';
    renderedButton = shallow(<Button value={mockValue} />);

    expect(renderedButton.find('.counter').length).toEqual(1);
  });

  it('selectData should send the value up when clicked', () => {
    renderedButton.simulate('click');
    expect(mockSelectData).toHaveBeenCalledWith(mockValue);
  });
});
