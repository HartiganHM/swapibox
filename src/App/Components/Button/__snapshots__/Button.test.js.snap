// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Tests Search should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    currentDisplay="People"
    selectData={[Function]}
    type="category button"
    value="People"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "People",
        " ",
        <div
          className="category-button-image-people"
/>,
      ],
      "className": "category button",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      "People",
      " ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "category-button-image-people",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "People",
          " ",
          <div
            className="category-button-image-people"
/>,
        ],
        "className": "category button",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        "People",
        " ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "category-button-image-people",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
